### PR TITLE
feat(core,phrases): add sso domains validation

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add the missing sso connectors domain config validation logic per our product design:

- throw error if any of the domain is in the public domain black list
- throw error if there are duplicate domains provided

## Testing

UT

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

**Review by darcyYe — COMMENTED** [review]
Can move the blacklist and the validation util methods to either `shared` or `toolkit` package so that console can also rely on it, instead of making another implementation. This could be easier for later maintainance.

File: packages/core/src/routes/sso-connector/utils.ts [hunk 1]
```diff
@@ -75,3 +79,58 @@ export const fetchConnectorProviderDetails = async (
...
+ *
+ * @param domains
+ * @returns
+ */
+export const validateConnectorDomains = (domains?: string[]) => {
+  if (!domains || domains.length === 0) {
+    return;
+  }
+
+  const blackListSet = new Set(singleSignOnDomainBlackList);
```
Comment (wangsijie) [inline packages/core/src/routes/sso-connector/utils.ts:96]: I've seen both "black list" and "block list", do we need to align this?

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/routes/sso-connector/utils.ts (hunk 1 above)
Comment (simeng-li) [reply packages/core/src/routes/sso-connector/utils.ts:96]: Yes, should align

**Comment by simeng-li** [comment]
> Can move the blacklist and the validation util methods to either `shared` or `toolkit` package so that console can also rely on it, instead of making another implementation. This could be easier for later maintainance.

aligned offline, use backend error data

**Pull request merged**